### DISCUSSION TITLE
Preventing the filed type 'userclass' from including "EVERYONE BUT"

### DISCUSSION THREAD

**Discussion by Vodhin** [e107inc/e107#5145] [Q&A Core] [answered]
I have a field set as "userclass" with "writeParms" =>  array ("classlist"=>"member,nobody") because I only want those options but e107 renders an EVERYONE BUT  set of options which defeats the purpose of having specific class options. How do I make the field only have "Members" and "Nobody"?

**Comment by tgtje** [comment]
**_Just opinion._**
EVERYONE BUT runs throughout the system, suppose forum had moderators. Those can be non members, but f.e. admins.
Having only members or nobody would prevent mods to access unless they become members too.
The main purpose is (always was) to allow admittance to all, EXCEPT a certain userclass...
I do not think this can be overridden.

**Reply by Vodhin** [reply]
I personally would not want to have a non-member (no USERID) be a moderator who could, for example, change or delete a forum post.

The plugin I am writing is for listing Real Estate properties and intended for an Admin to use (A Broker owns the website and has several Agents). I am in the process of extending the functionality to allow registered Members to submit a property as "For sale by owner" and I would like to use the Userclass selection to enable/disable the feature by letting the Admin choose either No One (disabled), Members, or a specific user class. I can easily make my own UC dropdown but figured I would try the e107 method first.

**Comment by Jimmi08** [comment, marked as answer]
Have you tried option "no-excludes" ?

**Reply by Vodhin** [reply]
Need syntax. I have tried:
`'writeParms' =>  array ('classlist'=>'member,nobody,no-excludes'),`
`'writeParms' =>  array ('classlist'=>'member,nobody','no-excludes'=>1)`
`'writeParms' =>  array ('classlist'=>'member,nobody','no-excludes'=>true)`
None of these worked.

**Reply by Vodhin** [reply]
Odd. even after empty cache and reload several times. I left GitHub, decided to reload that form the userclass select was on and bingo. 

`'writeParms' =>  array ('classlist'=>'member,nobody,no-excludes'),`

The above is correct. I now only see the classes I defined.

**Reply by Jimmi08** [reply]
this is working:
`$frm->userclass('news_editauthor', vartrue($pref['news_editauthor']),'off', 'nobody,member,no-excludes')."`

![image](https://github.com/e107inc/e107/assets/5429548/78f3f5ac-ca7b-4bc5-a840-35e6136c437c)

I can see you solved it :)

**Reply by Moc** [reply]
Will fix documentation for this. Thanks for letting us know! https://github.com/orgs/e107inc/projects/1/views/1?pane=issue&itemId=47238853
